explainer-server/app/views/explainEditor.scala.html

```diff
@@ -68,6 +66,24 @@
             }
         }
 
+        /*
+         * This function (and the accompanying handler) exists because there doesn't seem to exist a way
+         * to generate a checkbox using ScalaTags (ScalaJS's HTML generation library)
+         */
+        function installDisplayTypeTickBox(displayType){
+            console.log(displayType);
+            var checkedstring = displayType=='Expandable' ? ' checked ' : ''
+            $(".explainer-editor__displayType-inner").html('<input class="explainer-editor__displayType-checkbox" type="checkbox" name="displayType" value="long" '+checkedstring+'> Long explainer')

```

Comment:
@philmcmahon @ShaunYearStrong 

So I would like to sort of make a move on this...

**javascript versus scalatags**

Generating the input on scalaTags, is possible but the event handler then suffers from the same caching problem as before. In other words the following caches the state of the checkbox

```
    val checkbox: TypedTag[Input] = input(
      id:="mychecbbox",
      `type`:="checkbox"
    )
    val check = checkbox.render
    check.onchange = (e: dom.Event) => {
      g.console.log( check.value )
    }
```

So I will stick use scalaTags to generate the input but will use javascript to handle the event. 

**CSS classes versus ids**

Will follow CSS best practices. So I will use classes (as currently the case).

**DOM API**

Thanks for the suggestion. I will do that next time I need to generate DOM elements. I actually thought about it, but then didn't see a big deal with injecting the HTML. Bad Pascal... 

Cheers,

ps: see coming commit ...